lib/mail.ex

```diff
@@ -34,21 +34,36 @@ defmodule Mail do
 
   If a text part already exists this function will replace that existing
   part with the new part.
+
+  ## Options
+
+  * `:charset` - The character encoding standard for content type
   """
-  def put_text(%Mail.Message{multipart: true} = message, body) do
+  def put_text(message, body, opts \\ [])
+
+  def put_text(%Mail.Message{multipart: true} = message, body, opts) do
     message =
       case Enum.find(message.parts, &Mail.Message.match_body_text/1) do
         %Mail.Message{} = part -> Mail.Message.delete_part(message, part)
         _ -> message
       end
 
-    Mail.Message.put_part(message, Mail.Message.build_text(body))
+    Mail.Message.put_part(message, Mail.Message.build_text(body, opts))
   end
 
-  def put_text(%Mail.Message{} = message, body) do
+  def put_text(%Mail.Message{} = message, body, opts) do
+    content_type =
+      case opts do
+        charset: charset ->
+          ["text/plain", {"charset", charset}]
+
+        _else ->
+          "text/plain"
+      end
+
     Mail.Message.put_body(message, body)
     |> Mail.Message.put_header(:content_transfer_encoding, :quoted_printable)
-    |> Mail.Message.put_content_type("text/plain")
+    |> Mail.Message.put_content_type(content_type)
```

Comment:
If this is meant to be a text part why would we want the `content_type` to be overridable?